code/network/multimsgs.cpp

```diff
@@ -6889,11 +6889,12 @@ void send_asteroid_hit( object *objp, object *other_objp, vec3d *hitpos, float d
 {
 	int packet_size;
 	ubyte data[MAX_PACKET_SIZE], packet_type;
-	vec3d vec;
+	vec3d vec = vmd_zero_vector;
```

Comment:
It looks like you could avoid using `vec` at all, and just do e.g. `hitpos = &vmd_zero_vector`

Reply:
If you mean form the callers prespective, I had considered this, but I was a little bit worried about future-proofing I guess, and figured it would be safest to simply assume that callers might provide nullptr.

Reply:
I mean that the `vec3d vec` temporary variable isn't needed here, since if `hitpos` and/or `force` are null, they can be pointed to the zero vector.  But it's a trivial optimization.

Reply:
Oh nvm, I see now.